Source/WTF/wtf/SortedArrayMap.h

```diff
@@ -230,51 +227,51 @@ template<typename ArrayType> template<typename KeyArgument> inline bool SortedAr
     return iterator != std::end(m_array) && *iterator == *parsedKey;
 }
 
-constexpr int strcmpConstExpr(const char* a, const char* b)
+constexpr int compareSpansConstExpr(std::span<const char> a, std::span<const char> b)
 {
-    while (*a == *b && *a && *b) {
-        ++a;
-        ++b;
-    }
-    return *a == *b ? 0 : *a < *b ? -1 : 1;
+    auto commonLength = std::min(a.size(), b.size());
+    size_t i = 0;
+    while (i < commonLength && a[i] == b[i])
+        ++i;
+    auto aCharacter = i < a.size() ? a[i] : 0;
+    auto bCharacter = i < b.size() ? b[i] : 0;
+    return aCharacter == bCharacter ? 0 : aCharacter < bCharacter ? -1 : 1;
 }
 
-template<typename CharacterType> inline bool lessThanASCIICaseFolding(std::span<const CharacterType> characters, const char* literalWithNoUppercase)
+template<typename CharacterType> inline bool lessThanASCIICaseFolding(std::span<const CharacterType> characters, ASCIILiteral literalWithNoUppercase)
 {
-    for (auto character : characters) {
-        auto literalCharacter = *literalWithNoUppercase;
-        if (!literalCharacter)
-            return false;
+    size_t index = 0;
+    for (auto character : characters.first(std::min(characters.size(), literalWithNoUppercase.length()))) {
```

Comment:
I think you can use IndexedRange here, so you don't have to `++index` at the bottom of the loop.